flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/catalog/GenericInMemoryCatalogTest.java

```diff
@@ -52,6 +60,197 @@ static void init() {
         catalog.open();
     }
 
+    // TODO (FLINK-35020) : remove after implementing dropModel in catalog
+    @AfterEach
+    void cleanup() throws Exception {
+        if (catalog.modelExists(modelPath1)) {
+            ((GenericInMemoryCatalog) catalog).dropModel(modelPath1, true);
+        }
+        if (catalog.modelExists(modelPath2)) {
+            ((GenericInMemoryCatalog) catalog).dropModel(modelPath2, true);
+        }
+        super.cleanup();
+    }
+
+    // These are put here instead of CatalogTest class since model operations are not implemented
```

Comment:
rather introduce a `supportsModels(): boolean` method in CatalogTest and move those tests to the testbase. Set this flag to false and ignore them for Hive catalog.